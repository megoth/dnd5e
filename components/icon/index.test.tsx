import React from "react";
import { render } from "@testing-library/react";
import Icon from "./index";

describe("Icon", () => {
  it("renders icons", () => {
    expect(render(<Icon name="close" />).asFragment()).toMatchSnapshot();
    expect(render(<Icon name="menu" />).asFragment()).toMatchSnapshot();
  });
});
