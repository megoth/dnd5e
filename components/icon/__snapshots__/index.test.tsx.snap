// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders icons 1`] = `
<DocumentFragment>
  <svg
    class="icon"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18L18 6M6 6l12 12"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon renders icons 2`] = `
<DocumentFragment>
  <svg
    class="icon"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 16l-4-4m0 0l4-4m-4 4h14m-5 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h7a3 3 0 013 3v1"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon renders icons 3`] = `
<DocumentFragment>
  <svg
    class="icon"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 6h16M4 12h16M4 18h16"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon renders icons 4`] = `
<DocumentFragment>
  <svg
    class="icon"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 6V4m0 2a2 2 0 100 4m0-4a2 2 0 110 4m-6 8a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4m6 6v10m6-2a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</DocumentFragment>
`;
